sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -433,12 +454,14 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
                                    download_if_missing=download_if_missing)
 
     if os.path.exists(target_file):
-        X_train, X_test = joblib.load(target_file)
+        X_train, X_test, feature_names = joblib.load(target_file)
```

Comment:
we probably should handle the case that the file exists from an earlier version, but doesn't have the `feature_names`

Reply:
What would be the strategy there? I would almost advocate for asking to remove the cached file.

Reply:
I am okay with asking to remove as long as the full path is shown to the user.

Reply:
we could also version those files somehow so that it works with multiple versions of sklearn in the same home directory with the default paths. But I guess that's too much work.

As an alternative I'm happy to ask the user and remove while explaining why we need to remove it and that the new one works for sklearn>=0.xx